pkg/core/interop/context.go

```diff
@@ -392,23 +392,17 @@ func (c *HFSpecificContractMD) GetMethodByOffset(offset int) (HFSpecificMethodAn
 
 // GetMethod returns method `name` with the specified number of parameters.
 func (c *HFSpecificContractMD) GetMethod(name string, paramCount int) (HFSpecificMethodAndPrice, bool) {
-	index := sort.Search(len(c.Methods), func(i int) bool {
-		md := c.Methods[i]
-		res := strings.Compare(name, md.MD.Name)
-		switch res {
-		case -1, 1:
-			return res == -1
-		default:
-			return paramCount <= len(md.MD.Parameters)
+	index, ok := slices.BinarySearchFunc(c.Methods, HFSpecificMethodAndPrice{}, func(a, _ HFSpecificMethodAndPrice) int {
+		res := strings.Compare(a.MD.Name, name)
+		if res != 0 || paramCount == -1 {
+			return res
```

Comment:
Is this search behaviour deterministic from run to run? It should be, but we'd better be 100% sure. Consider the following case: malicious contract is deployed, it has two `verify` methods that differ by parameters count. We need the result of this search be deterministic from run to run and also matching C# node.

Reply:
I've restored the original behavior, but looks like C# does `Methods.FirstOrDefault` from the Manifest which is not sorted at all in general. But `HFSpecificContractMD` is only used for native contracts, they don't have a lot of surprises.

Reply:
You're right, I missed that it's for natives only.